packages/sync/src/utils/buildSyncLookupTraverseJoins.ts

```diff
@@ -0,0 +1,28 @@
+export function buildSyncLookupTraverseJoins(
+  tablesToTraverse: string[],
+  customForeignKeys?: Record<
+    string,
+    { fromTable: string; fromKey: string; toTable: string; toKey: string }
+  >,
+) {
+  return tablesToTraverse
+    .slice(1)
+    .map((table, i) => {
+      const previousTable = tablesToTraverse[i];
```

Comment:
I haven't seen this pattern before, it's a clever way to run through an array comparing each pair of contiguous records